bigchaindb/tendermint/core.py

```diff
@@ -121,3 +131,12 @@ def commit(self):
 
         data = self.block_txn_hash.encode('utf-8')
         return Result.ok(data=data)
+
+
+def cast_validator(v):
+    pub_key = v['pub_key']['data']
+    # NOTE: tendermint expects public to be ecoded in go-wire format
+    # so `01` has to be appended
+    pubKey = bytes.fromhex('01{}'.format(pub_key))
+    return Validator(pubKey=pubKey,
+                     power=v['power'])
```

Comment:
Why does nothing have to be appended before the power? I guess an integer is a basic type and doesn't need a prefix in go-wire encoding.

Reply:
Yup, but more specifically `power` is not expected to be encoded in go-wire format i.e. only `pubKey` is expected to be encoded in go-wire format.